core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/arrow/ArrowNltkSentiment.java

```diff
@@ -0,0 +1,274 @@
+            return false;
+        }
+
+        try {
+            if (Files.notExists(Paths.get(TO_PYTHON_FILE))) {
+                Files.createFile(Paths.get(TO_PYTHON_FILE));
+            }
+            new ChunkedArrowWriter<Tuple>(chunkSize, this::vectorizeTupleToPython).write(
+                    new File(TO_PYTHON_FILE), batchedTupleBuffer, tupleToPythonSchema);
+        } catch (IOException e) {
+            throw new DataflowException(e.getMessage(), e);
+        }
+        return true;
+    }
+
+    // Process the data file using NLTK
+    private void computeClassLabel() {
+        /*
+         *  In order to use the NLTK package to do classification, we start a
+         *  new process to run the package, and wait for the result of running
+         *  the process as the class label of this text field.
+         *  Python call format:
+         *      #python3 nltk_sentiment_classify picklePath dataPath resultPath
+         * */
+        try{
+            List<String> args = new ArrayList<String>(
+                    asList(PYTHON, PYTHON_SCRIPT, picklePath, ArrowNltkSentiment.TO_PYTHON_FILE, FROM_PYTHON_FILE));
+            ProcessBuilder processBuilder = new ProcessBuilder(args);
+
+            Process p = processBuilder.start();
+            p.waitFor();
+
+            //Read label result from arrow file generated by Python.
+            idClassMap = new HashMap<String, Integer>();
+
+            //Prepare for read
+            File arrowFile = new File(FROM_PYTHON_FILE);
+            FileInputStream fileInputStream = new FileInputStream(arrowFile);
+            SeekableReadChannel seekableReadChannel = new SeekableReadChannel(fileInputStream.getChannel());
+            ArrowFileReader arrowFileReader = new ArrowFileReader(seekableReadChannel,
+                    new RootAllocator(Integer.MAX_VALUE));
+            VectorSchemaRoot root  = arrowFileReader.getVectorSchemaRoot(); // get root
+            List<ArrowBlock> arrowBlocks = arrowFileReader.getRecordBlocks();
+
+            //For every block(arrow batch / or called 'chunk' here)
+            for (ArrowBlock rbBlock : arrowBlocks) {
+                if (!arrowFileReader.loadRecordBatch(rbBlock)) { // load the batch
+                    throw new IOException("Expected to read record batch");
+                }
+                List<FieldVector> fieldVector = root.getFieldVectors();
+                VarCharVector idVector = ((VarCharVector) fieldVector.get(0));
+                BigIntVector predVector = ((BigIntVector) fieldVector.get(1));
+                for (int j = 0; j < idVector.getValueCount(); j++) {
+                    String id = new String(idVector.get(j), StandardCharsets.UTF_8);
+                    int label = (int) predVector.get(j);
+                    try {
+                        idClassMap.put(id, label);
+                    } catch (NumberFormatException e) {
+                        idClassMap.put(id, 0);
```

Comment:
DO NOT just "eat" the exception and put a default value, this is a very very bad practice. If the exception happens, it means something goes wrong, and we would never know if something goes wrong. Just don't catch the exception, let it crash is the best way.

Reply:
Thank you, I've updated my code. But actually this "eating exception and put a default value" problem is inherited from the original [`NltkSentimentOperator` code](https://github.com/Texera/texera/blob/master/core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/nlp/sentiment/NltkSentimentOperator.java#L168). I did't want to modify the other behaviors at first so I followed the same practice.